src/UtilityTrait.php

```diff
@@ -446,4 +453,37 @@ private function intersectArrayRecursively($master, $mask)
         }
         return $master;
     }
+
+    /**
+     * Appends query param at the end of url.
+     *
+     * @param string $url
+     * @param string $varName
+     * @param string $value
+     */
+    function appendQueryParam(&$url, $varName, $value)
+    {
+        if (!isset($varName) || !isset($value)) {
+            return;
+        }
+
+        if (strpos($url, '?')) {
+            $url .= "&" . $varName . "=" . $value;
+        } else {
+            $url .= "?" . $varName . "=" . $value;
+        }
+    }
+
+    /**
+     * @param array $response
+     * @param string $lang
+     */
+    private function addLanguage(&$response, $lang)
+    {
+        if (isset($response['confirmLink'])) {
+            $this->appendQueryParam($response['confirmLink'], 'lang', $lang);
+        } else if (isset($response['authorizationLink'])) {
+            $this->appendQueryParam($response['confirmLink'], 'lang', $lang);
```

Comment:
This should be `authorizationLink`?
I think it'd be better if you would just call appendQueryParam method from initPayment/auth methods, i.e.:

```
$response['confirmLink'] = $this->appendQueryParam($response['confirmLink'], 'lang', $lang);
```

and

```
$response['authorizationLink'] = $this->appendQueryParam($response['authorizationLink'], 'lang', $lang);
```

also, please test as the the code with `authorizationLink` wouldn't have worked :)